src/app/(pages)/projects/layout.tsx

```diff
@@ -0,0 +1,21 @@
+"use client";
+ import { useQuery} from 'react-query';
+ import { getAllProjects } from '@/app/Service/ProjectServices';
+import ResponsiveDrawer from '@/app/components/drawer/Drawer';
+import { AiOutlineLoading3Quarters } from "react-icons/ai";
+import { BiError } from "react-icons/bi";
+
+export default function page({ children }: { children: React.ReactNode }) {
+   const { data: Allprojects , isLoading, error} = useQuery('Allprojects', getAllProjects);
+   if (isLoading) return <p>Loading projects <AiOutlineLoading3Quarters /> </p>;
+   if (error) return <p>Error loading projects <BiError /></p>;
+  return (
+    <>
+    {Allprojects && <ResponsiveDrawer projects={Allprojects} />}
```

Comment:
it not the good solution for this 
but use (use query) inside the Responsive Drawer it self and make in consideration that react query cache data and if use it in another place with same key will get data from cache not doing new request

Reply:
and after doing that change this layout to be server side that is more performant